src/participant_network.star

```diff
@@ -67,9 +70,115 @@ def launch_participant_network(
     global_tolerations,
     parallel_keystore_generation=False,
 ):
+    network_id = network_params.network_id
     num_participants = len(participants)
-    if network_params.network == constants.NETWORK_NAME.kurtosis:
-        # We are running a kurtosis network
+    latest_block = ""
+    cancun_time = 0
+    prague_time = 0
+    if (
+        network_params.network == constants.NETWORK_NAME.kurtosis
+        or constants.NETWORK_NAME.shadowfork in network_params.network
+    ):
+        if (
+            constants.NETWORK_NAME.shadowfork in network_params.network
+        ):  # shadowfork requires some preparation
+            base_network = network_params.network.split("-shadowfork")[
+                0
+            ]  # overload the network name to remove the shadowfork suffix
+            if constants.NETWORK_NAME.ephemery in base_network:
+                chain_id = plan.run_sh(
+                    run="curl -s https://ephemery.dev/latest/config.yaml | yq .DEPOSIT_CHAIN_ID | tr -d '\n'",
+                    image="linuxserver/yq",
+                )
+                network_id = chain_id.output
+            else:
+                network_id = constants.NETWORK_ID[
+                    base_network
+                ]  # overload the network id to match the network name
+            latest_block = plan.run_sh(  # fetch the latest block
+                run="mkdir -p /shadowfork && \
+                    curl -o /shadowfork/latest_block.json https://ethpandaops-ethereum-node-snapshots.ams3.digitaloceanspaces.com/"
+                + base_network
+                + "/geth/latest/latest_snapshot_block.json",
+                image="badouralix/curl-jq",
+                store=[StoreSpec(src="/shadowfork", name="latest_blocks")],
+            )
+
+            # maybe we can do the copy in the same step as the fetch?
+            for index, participant in enumerate(participants):
+                tolerations = input_parser.get_client_tolerations(
+                    participant.el_tolerations,
+                    participant.tolerations,
+                    global_tolerations,
+                )
+                cl_client_type = participant.cl_client_type
+                el_client_type = participant.el_client_type
+
+                # Zero-pad the index using the calculated zfill value
+                index_str = shared_utils.zfill_custom(
+                    index + 1, len(str(len(participants)))
+                )
+
+                el_service_name = "el-{0}-{1}-{2}".format(
+                    index_str, el_client_type, cl_client_type
+                )
+                shadowfork_data = plan.add_service(
+                    name="shadowfork-{0}".format(el_service_name),
+                    config=ServiceConfig(
+                        image="rclone/rclone:1.55.1",
+                        cmd=[
+                            "rclone copy -P mys3:ethpandaops-ethereum-node-snapshots/"
```

Comment:
i am surprised that this runs in parallel and doesn't wait for the r clone to finish. Its nice though.